tests/test_weighted_pandas.py

```diff
@@ -465,3 +442,163 @@ def test_WeightedSeries_nan():
+    dat = []
+    for _ in range(3000):
+        dat.append(x0)
+        x1, logL1 = logL(np.random.normal(x0, s/2))
+        if np.log(np.random.rand()) < logL1 - logL0:
+            x0, logL0 = x1, logL1
+
+    df = DataFrame(dat, columns=["x", "y", "z", "w"])
+    weights = df.groupby(df.columns.tolist(), sort=False).size().values
+    wdf = WeightedDataFrame(df.drop_duplicates(), weights=weights)
+    return df, wdf
+
+
+def test_WeightedDataFrame_hist():
+    df, wdf = mcmc_run()
+
+    df_axes = df.hist()
+    wdf_axes = wdf.hist()
+    for df_ax, wdf_ax in zip(df_axes.flatten(), wdf_axes.flatten()):
+        for df_patch, wdf_patch in zip(df_ax.patches, wdf_ax.patches):
+            assert df_patch.get_height() == wdf_patch.get_height()
+            assert df_patch.get_width() == wdf_patch.get_width()
+            assert df_patch.get_xy() == wdf_patch.get_xy()
+
+    df_axes = df.plot.hist(subplots=True)
+    wdf_axes = wdf.plot.hist(subplots=True)
+    for df_ax, wdf_ax in zip(df_axes.flatten(), wdf_axes.flatten()):
+        for df_patch, wdf_patch in zip(df_ax.patches, wdf_ax.patches):
+            assert df_patch.get_height() == wdf_patch.get_height()
+            assert df_patch.get_width() == wdf_patch.get_width()
+            assert df_patch.get_xy() == wdf_patch.get_xy()
+
+    plt.close("all")
+
+
+def test_WeightedSeries_hist():
+    df, wdf = mcmc_run()
+
+    fig, axes = plt.subplots(2)
+    df.x.hist(ax=axes[0])
+    wdf.x.hist(ax=axes[1])
+
+    for df_patch, wdf_patch in zip(axes[0].patches, axes[1].patches):
+        assert df_patch.get_height() == wdf_patch.get_height()
+        assert df_patch.get_width() == wdf_patch.get_width()
+        assert df_patch.get_xy() == wdf_patch.get_xy()
+
+    fig, axes = plt.subplots(2)
+    df.x.plot.hist(ax=axes[0])
+    wdf.x.plot.hist(ax=axes[1])
+
+    for df_patch, wdf_patch in zip(axes[0].patches, axes[1].patches):
+        assert df_patch.get_height() == wdf_patch.get_height()
+        assert df_patch.get_width() == wdf_patch.get_width()
+        assert df_patch.get_xy() == wdf_patch.get_xy()
+
+    plt.close("all")
+
+
+def test_KdePlot():
+    df, wdf = mcmc_run()
+
+    bw_method = 0.3
+    fig, axes = plt.subplots(2)
+    df.x.plot.kde(bw_method=bw_method, ax=axes[0])
+    wdf.x.plot.kde(bw_method=bw_method, ax=axes[1])
+    df_line, wdf_line = axes[0].lines[0], axes[1].lines[0]
+    assert (df_line.get_xdata() == wdf_line.get_xdata()).all()
+    assert_allclose(df_line.get_ydata(),  wdf_line.get_ydata(), atol=1e-4)
+
+    plt.close("all")
+
+
+def test_scatter_matrix():
+    df, wdf = mcmc_run()
+    np.random.seed(0)
+
+    scatter_matrix(df)
+    axes = scatter_matrix(wdf)
+    n = len(axes[0, 1].collections[0].get_offsets().data)
+    neff = channel_capacity(wdf.weights)
+    assert_allclose(n, neff, atol=np.sqrt(n))
+
+    plt.close("all")
+
+
+def test_bootstrap_plot():
+    df, wdf = mcmc_run()
+    bootstrap_plot(wdf.x)
+    plt.close("all")
+
+
+def test_BoxPlot():
```

Comment:
The boxplot 'tests' don't actually test for anything, since it's hard to extract the values of the weighted boxplots for comparison with the unweighted version (and they give slightly different answers).